### PR TITLE
MH-13487 Initialize this repository

### PR DESCRIPTION
**Pull request by pmiddend**
[MH-13483](https://opencast.jira.com/browse/MH-13483) introduces basic statistics visualization facilities into Opencast. In particular, a statistics provider implementation for InfluxDB 1.x time series is provided, i.e. Opencast can visualize InfluxDB time series data. 

The Opencast InfluxDB Adapter (this pull request) is implemented as a standalone Java application that continuously parses the Apache log files HTTP access logs to gather data about the usage of assets (here: "Views" of video files) and feeds those info into InfluxDB.

The adapter optionally uses the External API to find out to which series a given event belongs so that view data can be aggregated on series level, too.

Currently implemented feature set:
- Both single- and multi-tenant support
- External API integration to retrieve additional metadata from Opencast (series)
- Dynamic time window analysis of HTTP accesses to heuristically determine "number of views" (access to the same resource within a given time window don't count as multiple views)
- Some filter options to ignore requests that should not be counted

Current limitations:
- The adapter is implemented for a specific Apache log format. In theory, it could be generalised to support other log formats, web servers or even to a general line-based log analyzer. This is, however, not in the scope of this work.

This work is sponsored by SWITCH.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+target/
+/build/opencast-influxdb-adapter-1.0.jar
```
Comment (gregorydlogan) [inline .gitignore]: Wouldn't ignoring `build` make more sense, rather than the jar + version?

Comment (pmiddend) [reply .gitignore]: Oh, absolutely, I just missed that. Will be fixed soon.

**Comment by gregorydlogan** [comment]
@wsmirnow @staubesv I'm assuming that Plapadoo and co should have write access to this repo.  I've given @doofy and @pmiddend write for this repo.  Anyone else needed?

**Comment by wsmirnow** [comment]
Hi Greg,
you can grant write permissions to all Opencast developers. But it will
take a while. Doing it on demand is also an option.
No really, I havn't a strong opinion about that. It is ok, if the main
developer have write access to this repo.

Regards
Waldemar

ELAN e.V.
Heger-Tor-Wall 12
49074 Osnabrück - Germany
Tel: +49 541 / 969 6533
Web: https://elan-ev.de

Am 03.05.19 um 17:51 schrieb Greg Logan:

**Comment by staubesv** [comment]
@gregorydlogan I would say that all Opencast committers should have write access to this repo.

**Comment by gregorydlogan** [comment]
@staubesv @wsmirnow committers already have write access, but @pmiddend didn't.  @doofy already did, but I forgot he was a committer :)

**Comment by staubesv** [comment]
@wsmirnow What is the status of this review?

**Comment by wsmirnow** [comment]
Looks fine for me. I havn't run this code but the build was successfull.

**Comment by staubesv** [comment]
Thanks for the review!

**Pull request merged**